webapp/app/forms/fields.py

```diff
@@ -63,6 +63,46 @@ def _value(self):
         return self.data.split('-') if self.data else ''
 
 
+class IdNrWidget(MultipleInputFieldWidget):
+    """A divided input field with four text input fields, limited to two to three chars."""
+    sub_field_separator = ''
+    input_field_lengths = [2, 3, 3, 3]
+
+
+class IdNrField(SteuerlotseStringField):
+    def __init__(self, label='', validators=None, **kwargs):
+        super(IdNrField, self).__init__(label, validators, **kwargs)
+        self.widget = IdNrWidget()
+
+    def process_formdata(self, valuelist):
+        if valuelist:
+            self.data = valuelist
+        elif self.data is None:
+            self.data = []
+
+    def _value(self):
+        # In case the validation was not successful, we already have the data as a list of values.
+        if isinstance(self.data, list):
+            return self.data
+
+        # Once the validation has gone through, post_validate() stores the data as string.
+        # As we know that it is correct, we can just separate it in chunks here.
+        split_data = []
+        chunk_sizes = self.widget.input_field_lengths
+        current_idx = 0
+        for chunk_size in chunk_sizes:
+            if self.data:
+                split_data.append(self.data[current_idx: current_idx + chunk_size])
+            current_idx += chunk_size
```

Comment:
```suggestion
        start_idx = 0
        for chunk_size in chunk_sizes:
            end_index = start_idx + chunk_size
            if self.data:
                split_data.append(self.data[start_idx: end_index])
            start_idx = end_index
```

I was just thinking how to make it more readable. However, it is not that important